File: frontend/src/types/annotation.ts
export interface Category {
  id: string
  video_id: string
  name: string
  color: string
  created_at: string
}

export interface Annotation {
  id: string
  video_id: string
  frame_number: number
  timestamp_ms: number
  label: string
  category_id?: string
  category?: Category
  created_at: string
  updated_at: string
}

export interface CreateAnnotationRequest {
  frame_number: number
  label: string
  category_id?: string
}

export interface UpdateAnnotationRequest {
  frame_number: number
  label: string
}

File: backend/app/schemas/annotation.py
from typing import Optional
from pydantic import BaseModel


class AnnotationCreate(BaseModel):
    frame_number: int
    label: str = ""
    category_id: Optional[str] = None


class AnnotationRead(BaseModel):
    id: str
    video_id: str
    frame_number: int
    timestamp_ms: float
    label: str
    category_id: Optional[str] = None
    created_at: str
    updated_at: str


class BulkCreate(BaseModel):
    start_frame: int
    end_frame: int
    count: int
    prefix: str = ""
    category_id: Optional[str] = None


class ShiftRequest(BaseModel):
    offset_ms: float
